src/libos/src/net/syscalls.rs

```diff
@@ -666,23 +695,26 @@ pub fn do_epoll_wait(
     };
 
     // A new vector to store EpollEvent, which may degrade the performance due to extra copy.
-    let mut inner_events: Vec<EpollEvent> =
-        vec![EpollEvent::new(EpollEventFlags::empty(), 0); max_events];
+    let mut inner_events: Vec<MaybeUninit<EpollEvent>> = vec![MaybeUninit::uninit(); max_events];
 
     debug!(
         "epoll_wait: epfd: {}, len: {:?}, timeout: {}",
         epfd,
         raw_events.len(),
-        timeout
+        timeout_ms,
     );
 
     let epfile_ref = current!().file(epfd as FileDesc)?;
-    let epoll_file = epfile_ref.as_epfile()?;
-
-    let count = epoll_file.wait(&mut inner_events, timeout)?;
+    let epoll_file = epfile_ref.as_epoll_file()?;
+    let timeout = if timeout_ms >= 0 {
+        Some(Duration::from_millis(timeout_ms as u64))
+    } else {
+        None
```

Comment:
“timeout of -1 causes epoll_wait() to block indefinitely”
Should we check the timeout?